src/imap/mod.rs

```diff
@@ -550,32 +552,74 @@ impl Imap {
         Ok(())
     }
 
-    /// return Result with (uid_validity, last_seen_uid) tuple.
+    /// return Result with (uid_validity, last_seen_uid, new_emails) tuple.
+    /// If in doubt, returns new_emails=true so emails are fetched.
     pub(crate) async fn select_with_uidvalidity(
         &mut self,
         context: &Context,
         folder: &str,
-    ) -> Result<(u32, u32)> {
-        self.select_folder(context, Some(folder)).await?;
+    ) -> Result<(u32, u32, bool)> {
```

Comment:
at this point I would suggest using a struct for the return type

Reply:
In the further process of the PR I will probably remove the first `u32` and maybe the second so let's wait until then

Reply:
Marking it as resolved, as this function only returns a single `bool` now.